components/api_request_helper/api_request_helper.cc

```diff
@@ -61,20 +63,60 @@ void APIRequestHelper::Request(
   if (max_body_size == -1u) {
     iter->get()->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
         url_loader_factory_.get(),
-        base::BindOnce(&APIRequestHelper::OnResponse, base::Unretained(this),
-                       iter, std::move(callback)));
+        base::BindOnce(&APIRequestHelper::OnResponse,
+                       weak_ptr_factory_.GetWeakPtr(), iter,
+                       std::move(callback), std::move(conversion_callback)));
   } else {
     iter->get()->DownloadToString(
         url_loader_factory_.get(),
-        base::BindOnce(&APIRequestHelper::OnResponse, base::Unretained(this),
-                       iter, std::move(callback)),
+        base::BindOnce(&APIRequestHelper::OnResponse,
+                       weak_ptr_factory_.GetWeakPtr(), iter,
+                       std::move(callback), std::move(conversion_callback)),
         max_body_size);
   }
 }
 
+void APIRequestHelper::DefaultConversionHandler(std::string* raw_response) {
+  // Used for requests that not requre conversions.
+  // Each consumer should use own conversion handler if necessary
+}
+
+void APIRequestHelper::Request(
+    const std::string& method,
+    const GURL& url,
+    const std::string& payload,
+    const std::string& payload_content_type,
+    bool auto_retry_on_network_change,
+    ResultCallback callback,
+    const base::flat_map<std::string, std::string>& headers /* ={} */,
+    size_t max_body_size /* =-1 */) {
+  RequestWithConversion(
+      method, url, payload, payload_content_type, auto_retry_on_network_change,
+      std::move(callback), headers,
+      base::BindOnce(&APIRequestHelper::DefaultConversionHandler,
+                     weak_ptr_factory_.GetWeakPtr()),
```

Comment:
In this case we should just use nullptr for conversion callback, then it won't be used in line 143-144 below since it's null.